data/tutorials/language/1ms_00_modules.md

```diff
@@ -1,545 +1,436 @@
-    val filteri : (int -> 'a -> bool) -> 'a t -> 'a t
-    val partition : ('a -> bool) -> 'a t -> 'a t * 'a t
-    val partition_map : ('a -> ('b, 'c) Either.t) -> 'a t -> 'b t * 'c t
-    val assoc : 'a -> ('a * 'b) t -> 'b
-    val assoc_opt : 'a -> ('a * 'b) t -> 'b option
-    val assq : 'a -> ('a * 'b) t -> 'b
-    val assq_opt : 'a -> ('a * 'b) t -> 'b option
-    val mem_assoc : 'a -> ('a * 'b) t -> bool
-    val mem_assq : 'a -> ('a * 'b) t -> bool
-    val remove_assoc : 'a -> ('a * 'b) t -> ('a * 'b) t
-    val remove_assq : 'a -> ('a * 'b) t -> ('a * 'b) t
-    val split : ('a * 'b) t -> 'a t * 'b t
-    val combine : 'a t -> 'b t -> ('a * 'b) t
-    val sort : ('a -> 'a -> int) -> 'a t -> 'a t
-    val stable_sort : ('a -> 'a -> int) -> 'a t -> 'a t
-    val fast_sort : ('a -> 'a -> int) -> 'a t -> 'a t
-    val sort_uniq : ('a -> 'a -> int) -> 'a t -> 'a t
-    val merge : ('a -> 'a -> int) -> 'a t -> 'a t -> 'a t
-    val to_seq : 'a t -> 'a Seq.t
-    val of_seq : 'a Seq.t -> 'a t
-    val optmap : ('a -> 'b option) -> 'a t -> 'b t
-  end
+module List = struct
+  include Stdlib.List
+  let uncons = function
+    | [] -> None
+    | hd :: tl -> Some (hd, tl)
+end
 ```
 
-It creates a module `Extensions.List` that has everything the standard `List`
-module has, plus a new `optmap` function. From another file, all we have to do
-to override the default `List` module is `open Extensions` at the beginning of
-the `.ml` file:
+It creates a module `Extlib.List` that has everything the standard `List` module
+has, plus a new `uncons` function. In order to override the default `List`
+module from another `.ml` file, we need to add `open Extlib` at the beginning.
 
-<!-- $MDX skip -->
+## Stateful Modules
+
+A module may have an internal state. This is the case for the `Random` module from the standard library. The functions `Random.get_state` and `Random.set_state` provide read and write access to the internal state, which is nameless and has an abstract type.
 ```ocaml
-open Extensions
+# let s = Random.get_state ();;
+val s : Random.State.t = <abstr>
+
+# Random.bits ();;
+- : int = 89809344
 
-...
+# Random.bits ();;
+- : int = 994326685
 
-List.optmap ...
+# Random.set_state s;;
+- : unit = ()
+
+# Random.bits ();;
+- : int = 89809344
 ```
+
+Values returned by `Random.bits` will differ when you run this code. The first
+and third calls return the same results, showing that the internal state was
+reset.
+
+## Conclusion
+
+In OCaml, modules are the basic means of organising software. To sum up, a module is a collection of definitions wrapped under a name. These definitions can be submodules, which allows the creation of hierarchies of modules. Top-level modules must be files and are the units of compilation. Every module has an interface, which is the list of definitions a module exposes. By default, a module's interface exposes all its definitions, but this can be restricted using the interface syntax.
```

Comment:
```suggestion
In OCaml, modules are the basic means of organising software. To sum up, a module is a collection of definitions wrapped under a name. These definitions can be submodules, which allow the creation of module hierarchies. Top-level modules must be files and are the units of compilation. Every module has an interface, the list of definitions a module exposes. By default, a module's interface exposes all its definitions, but this can be restricted using the interface syntax.
```

Reply:
Yes, that's better